spec/rspec/mocks/matchers/have_received_spec.rb

```diff
@@ -141,9 +141,41 @@ module Mocks
           let(:the_dbl) { double(:expected_method => nil) }
 
           before do
-            the_dbl.expected_method
-            the_dbl.expected_method
-            the_dbl.expected_method
+            the_dbl.expected_method(:one)
+            the_dbl.expected_method(:two)
+            the_dbl.expected_method(:one)
+          end
+
+          context "when constrained by `with`" do
+            it 'only considers the calls with matching args' do
+              expect(the_dbl).to have_received(:expected_method).with(:one).twice
+              expect(the_dbl).to have_received(:expected_method).with(:two).once
+            end
+
+            context "when the message is received without any args matching" do
+              it 'includes unmatched args in the error message' do
+                expect {
+                  expect(the_dbl).to have_received(:expected_method).with(:three).once
+                }.to raise_error(/#{Regexp.escape("expected: (:three)")}.*#{
+                                    Regexp.escape("got: ([:one, :two])")}/m)
```

Comment:
This doesn't read very well, if you see @myronmarston's specs in #841, you'll see you can compose this message using `a_string_including` with the two desired fragments, I'd prefer if you refactored these specs to use that pattern